### PR TITLE
Update github action

### PR DESCRIPTION
**Pull request by XiongKezhi**
Removed mac-os and included JDK11, code coverage

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -13,16 +11,32 @@ jobs:
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven
+
       - name: Build with Maven
-        run: mvn -V -ntp clean verify --file pom.xml '-Dgpg.skip'
+        run: mvn -Penable-jacoco clean verify -B -V -ntp
```
Comment (timja) [inline .github/workflows/maven.yml:37]: does a profile need to be added for this?

Comment (XiongKezhi) [reply .github/workflows/maven.yml:37]: Yes, in order to generate the jacoco report. It's defined in the original jenkins-plugin pom

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -13,16 +11,32 @@ jobs:
...
       - uses: actions/checkout@v2
-      - name: Set up JDK 1.8
+
+      - name: Set up JDK ${{ matrix.java }}
         uses: actions/setup-java@v1
         with:
-          java-version: 1.8
+          java-version: ${{ matrix.java }}
+
+      - name: Cache dependencies
```
Comment (uhafner) [inline .github/workflows/maven.yml:28]: On what level is this cache working? Is this cache shared between the matrix elements?

Comment (XiongKezhi) [reply .github/workflows/maven.yml:28]: Not shared between matrix elements, just cached for one single os's builds. See https://github.com/actions/cache#cache

Comment (uhafner) [reply .github/workflows/maven.yml:28]: Why do we need a cache then? There is only one step on a node?

Comment (XiongKezhi) [reply .github/workflows/maven.yml:28]: "A workflow can access and restore a cache created in the current branch, the base branch (including base branches of forked repositories), or the default branch (usually master)." from [restrictions-for-accessing-a-cache](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache).

So if I understand this correctly, the cache can be shared among different branches, thus speed up the action.

Comment (timja) [reply .github/workflows/maven.yml:28]: It’s to cache dependencies and make the build a bit faster

Comment (XiongKezhi) [reply .github/workflows/maven.yml:28]: > 
> 
> Not shared between matrix elements, just cached for one single os's builds. See https://github.com/actions/cache#cache

Maybe I should say "cached for the same os's builds among actions in different branches"

**Pull request merged**